configure.ac

```diff
@@ -1669,7 +1631,7 @@ case "$with_pmilib" in
     fi
     ;;
   cray)  
-    # set CPPFLAGS and LDFLAGS
+    # deprecate
```

Comment:
@RaymondMichael We are leaving this option just in case there are users want to build MPICH on older Cray systems. The options are not visible in `configure --help` and the `deprecate` keyword should dissuade users to use `--with-pmilib=cray` unless they know what they are doing.